README.md

```diff
@@ -1,109 +1,98 @@
-[![GitHub stars](https://img.shields.io/github/stars/codeigniter4/CodeIgniter4)](https://packagist.org/packages/codeigniter4/framework)
-[![GitHub license](https://img.shields.io/github/license/codeigniter4/CodeIgniter4)](https://github.com/codeigniter4/CodeIgniter4/blob/develop/LICENSE)
-[![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/codeigniter4/CodeIgniter4/pulls)
-<br>
+[![Static Analysis](https://github.com/codeigniter4/CodeIgniter4/actions/workflows/test-phpstan.yml/badge.svg)](https://github.com/codeigniter4/CodeIgniter4/actions/workflows/test-phpstan.yml)
+[![License](https://img.shields.io/github/license/codeigniter4/CodeIgniter4)](LICENSE)
 
-## What is CodeIgniter?
+CodeIgniter 4 is a modern, lightweight PHP full-stack framework focused on developer productivity, speed, and security. It provides a collection of expressive libraries, a powerful CLI, and an intuitive MVC structure that helps you ship production-ready applications quickly.
 
-CodeIgniter is a PHP full-stack web framework that is light, fast, flexible and secure.
-More information can be found at the [official site](https://codeigniter.com).
+---
 
-This repository holds the source code for CodeIgniter 4 only.
-Version 4 is a complete rewrite to bring the quality and the code into a more modern version,
-while still keeping as many of the things intact that has made people love the framework over the years.
+## Overview
 
-More information about the plans for version 4 can be found in [CodeIgniter 4](https://forum.codeigniter.com/forumdisplay.php?fid=28) on the forums.
+- **Lightweight core** with optional packages so you only ship what you need.
+- **Secure by design** thanks to CSRF protection, content security features, and built-in validation.
+- **Developer friendly** CLI (`php spark`) for scaffolding, migrations, tasks, and testing.
+- **Flexible architecture** that embraces PSR standards, Composer autoloading, and namespacing.
 
-### Documentation
+## Requirements
 
-The [User Guide](https://codeigniter.com/user_guide/) is the primary documentation for CodeIgniter 4.
+- PHP 8.1 or newer
+- `ext-intl` and `ext-mbstring` enabled
+- Recommended: `ext-curl`, `ext-json`, and database-specific extensions (`ext-mysqli`, `ext-pgsql`, etc.)
 
-You will also find the [current **in-progress** User Guide](https://codeigniter4.github.io/CodeIgniter4/).
-As with the rest of the framework, it is a work in progress, and will see changes over time to structure, explanations, etc.
+> PHP 8.1 reaches end of life on December 31, 2025. Upgrade sooner to stay supported.
 
-You might also be interested in the [API documentation](https://codeigniter4.github.io/api/) for the framework components.
+## Quick Start
 
-## Important Change with index.php
+1. **Clone and install dependencies**
+   ```bash
+   git clone https://github.com/codeigniter4/CodeIgniter4.git
+   cd CodeIgniter4
+   composer install
+   ```
```

Comment:
This step is for contributors only to the framework development. As such, this is not the go to way of installing this.